src/github/pullRequestGitHelper.ts

```diff
@@ -111,6 +120,79 @@ export class PullRequestGitHelper {
 		}
 	}
 
+	static async getBranchNRemoteForPullRequest(repository: Repository, pullRequest: PullRequestModel): Promise<{
+		branch: string,
+		remote?: string,
+		createdForPullRequest?: boolean,
+		remoteInUse?: boolean
+	} | null> {
+		let key = PullRequestGitHelper.buildPullRequestMetadata(pullRequest);
+		let configs = await repository.getConfigs();
+
+		let branchInfos = configs.map(config => {
+			let matches = PullRequestBranchRegex.exec(config.key);
+			return {
+				branch: matches && matches.length ? matches[1] : null,
+				value: config.value
+			};
+		}).filter(c => c.branch && c.value === key);
```

Comment:
if we had a BranchInfo type where branch is required we could do ((c: any): c is BranchInfo => ...) to avoid having to use !